### PR TITLE
ELY-1284: Allow ServiceLoaderSupplier to load services in a privileged block

### PR DESCRIPTION
**Pull request by yersan**
Jira issues:
https://issues.jboss.org/browse/ELY-1284
https://issues.jboss.org/browse/JBEAP-12111

### CODE REVIEW

File: src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java [hunk 1]
```diff
@@ -38,31 +41,43 @@
     private final ClassLoader classLoader;
     private int hashCode;
     private volatile E[] result;
+    private final AccessControlContext acc;
 
     public ServiceLoaderSupplier(final Class<E> service, final ClassLoader classLoader) {
         this.service = service;
         this.classLoader = classLoader;
+        this.acc = (System.getSecurityManager() != null) ? AccessController.getContext() : null;
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java]: The context should always be captured here, as the security manager might be initialized *after* one or more configurations are established.

File: src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java [hunk 2]
```diff
@@ -38,31 +41,43 @@
...
-                    for (;;) try {
-                        if (! iterator.hasNext()) {
-                            return (result = list.toArray(Arrays2.createArray(service, list.size()))).clone();
-                        }
-                        list.add(iterator.next());
-                    } catch (ServiceConfigurationError ignored) {}
+                    if (acc == null){
+                        result = loadServices(service, classLoader);
+                    } else {
+                        result = AccessController.doPrivileged((PrivilegedAction<E[]>) () -> loadServices(service, classLoader), acc);
```
Comment (dmlloyd) [inline src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java]: The security manager check could be ported here.

File: src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java (hunk 1 above)
Comment (yersan) [inline src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java]: @dmlloyd Fixed

File: src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java (hunk 2 above)
Comment (yersan) [inline src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java]: @dmlloyd Now I'm WildFlySecurityManager.isChecking() here

**Pull request merged**